tests/integration/test_charm.py

```diff
@@ -28,17 +30,67 @@ async def test_flask_is_up(
         assert "Hello, World!" in response.text
 
 
-@pytest.mark.parametrize("update_config", [{"webserver_timeout": 7}], indirect=True)
+@pytest.mark.parametrize(
+    "update_config, timeout",
+    [({"webserver_timeout": 7}, 7), ({"webserver_timeout": 5}, 5), ({"webserver_timeout": 3}, 3)],
+    indirect=["update_config"],
+    ids=["timeout-7", "timeout-5", "timeout-3"],
+)
 @pytest.mark.usefixtures("update_config")
-async def test_flask_webserver_config(
-    flask_app: Application, get_unit_ips: typing.Callable[[str], typing.Awaitable[tuple[str, ...]]]
+async def test_flask_webserver_timeout(
+    flask_app: Application,
+    get_unit_ips: typing.Callable[[str], typing.Awaitable[tuple[str, ...]]],
+    timeout,
 ):
     """
     arrange: build and deploy the flask charm, and change the gunicorn timeout configuration.
     act: send long-running requests to the flask application managed by the flask charm.
     assert: the gunicorn should restart the worker if the request duration exceeds the timeout.
     """
+    safety_timeout = timeout + 3
     for unit_ip in await get_unit_ips(flask_app.name):
-        assert requests.get(f"http://{unit_ip}:8000/sleep?duration=6", timeout=10).ok
+        assert requests.get(
+            f"http://{unit_ip}:8000/sleep?duration={timeout - 1}", timeout=safety_timeout
+        ).ok
         with pytest.raises(requests.ConnectionError):
-            requests.get(f"http://{unit_ip}:8000/sleep?duration=8", timeout=10)
+            requests.get(
+                f"http://{unit_ip}:8000/sleep?duration={timeout + 1}", timeout=safety_timeout
```

Comment:
What is the reason for having multiple of these checks?

Reply:
The two requests (timeout+1, timeout-1) is to probe the upper bound and lower bound for the timeout value.